support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionPurchaseBuilder.scala

```diff
@@ -48,35 +40,29 @@ class DigitalSubscriptionPurchaseBuilder(
     val todaysDate = today()
     val contractAcceptanceDate = todaysDate.plusDays(contractAcceptanceDelay)
 
-    val subscriptionData = buildProductSubscription(
-      requestId,
-      productRatePlanId,
-      contractAcceptanceDate = contractAcceptanceDate,
-      contractEffectiveDate = todaysDate,
-      readerType = digitalPack.readerType,
-      autoRenew = autoRenew,
-      initialTerm = initialTerm
-    )
-
-    val withRedemptionCode = addRedemptionCodeIfGift(digitalPack.readerType, giftCodeGenerator.generateCode(billingPeriod), subscriptionData)
-    val withPromoApplied = applyPromoCodeIfPresent(promotionService, maybePromoCode, country, productRatePlanId, withRedemptionCode)
-
-    EitherT.fromEither[Future](withPromoApplied)
-  }
+    val maybeRedemptionCode = (digitalPack.readerType, maybeGiftCode) match {
+      case (Gift, Some(giftCode)) => Some(giftCode)
+      case (Gift, _) | (_, Some(_)) => None // FIXME coding error - possible unredeemable sub
```

Comment:
I think we should probably explode at this point - the types we have defined are not good enough to make this impossible, but it's impossible